lib/handlers/allow.js

```diff
@@ -108,20 +125,31 @@ function documentFetcher (host, ldp, baseUri) {
  * - From `On-Behalf-Of` header (if it's present AND secretary is authorized)
  * - From `session.userId` (if no delegation header, or secretary unauthorized)
  *
- * @method getUserId
+ * @method getRequestingWebId
  * @param webId {string} Current user, from session.userId
  * @param onBehalfOf {string} Contents of the `On-Behalf-Of` delegation header
  * @param fetchDocument {Function} Result of `documentFetcher()` call
  * @return {Promise<string>} The user id (either the current user, or
  *   the user being delegated for).
  */
-function getUserId (webId, onBehalfOf, fetchDocument) {
-  let principal = onBehalfOf
-  if (!webId || !principal) {
+function getRequestingWebId (webId, onBehalfOf, fetchDocument) {
```

Comment:
Some procedure has to know where that data comes from, and rather than have every single caller of this function reach into the request object and pull out those values, it seems more maintainable to keep that logic here.